reference/matrix/coo_kernels.cpp

```diff
@@ -157,6 +159,75 @@ void convert_row_idxs_to_ptrs(std::shared_ptr<const ReferenceExecutor> exec,
 }
 
 
+template <typename ValueType, typename IndexType>
+void mem_size_bccoo(std::shared_ptr<const DefaultExecutor> exec,
+                    const matrix::Coo<ValueType, IndexType>* coo,
+                    IndexType* rows, IndexType* offsets,
+                    const size_type num_blocks, const size_type block_size,
+                    size_type* mem_size)
+{
+    // Computation of rows, offsets and m (mem_size)
+    const IndexType* row_idxs = coo->get_const_row_idxs();
+    const IndexType* col_idxs = coo->get_const_col_idxs();
+    const ValueType* values = coo->get_const_values();
+    const size_type num_rows = coo->get_size()[0];
+    const size_type num_stored_elements = coo->get_num_stored_elements();
+    size_type nblk = 0, blk = 0, rowR = 0, colR = 0, shf = 0;
+    for (size_type i = 0; i < num_stored_elements; i++) {
+        const size_type row = row_idxs[i];
+        const size_type col = col_idxs[i];
+        const ValueType val = values[i];
+        cnt_detect_newblock(nblk, shf, rowR, row - rowR, colR);
+        size_type colRS =
+            cnt_position_newrow_mat_data(row, col, shf, rowR, colR);
+        cnt_next_position_value(colRS, shf, colR, val, nblk);
+        cnt_detect_endblock(block_size, nblk, blk);
+    }
+
+    *mem_size = shf;
+}
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
+    GKO_DECLARE_MEM_SIZE_BCCOO_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void convert_to_bccoo(std::shared_ptr<const ReferenceExecutor> exec,
+                      const matrix::Coo<ValueType, IndexType>* source,
+                      matrix::Bccoo<ValueType, IndexType>* result)
+{
+    size_type block_size = result->get_block_size();
+    IndexType* rows_data = result->get_rows();
+    IndexType* offsets_data = result->get_offsets();
+    uint8* chunk_data = result->get_chunk();
+
+    // Computation of chunk
+    const IndexType* row_idxs = source->get_const_row_idxs();
+    const IndexType* col_idxs = source->get_const_col_idxs();
+    const ValueType* values = source->get_const_values();
+    const size_type num_rows = source->get_size()[0];
+    const size_type num_stored_elements = source->get_num_stored_elements();
+    size_type nblk = 0, blk = 0, rowR = 0, colR = 0, shf = 0;
+    if (num_stored_elements > 0) {
+        offsets_data[0] = 0;
+    }
+    for (size_type i = 0; i < num_stored_elements; i++) {
+        const size_type row = row_idxs[i];
+        const size_type col = col_idxs[i];
+        const ValueType val = values[i];
+        put_detect_newblock(chunk_data, rows_data, nblk, blk, shf, rowR,
+                            row - rowR, colR);
```

Comment:
In ginkgo, we use `snake_name` for variable from https://github.com/ginkgo-project/ginkgo/wiki/Contributing-guidelines#variables

Reply:
I think you propose to change rowR by row_xxx. Is it ok?

Reply:
Yes, it is okay